runtime/globroots.c

```diff
@@ -27,9 +27,17 @@
 
 static caml_plat_mutex roots_mutex = CAML_PLAT_MUTEX_INITIALIZER;
 
+/* Greater than zero when the current thread is scanning the roots */
+static CAMLthread_local int iterating_roots = 0;
```

Comment:
Was there a particular reason this takes values other than 0 and 1?

Reply:
No, I'm fairly sure it's only ever 0 or 1, but it was easier to be robust to the possibility of higher numbers than to verify that they definitely never occur.